### ISSUE TITLE
HashLock Transaction should accept aliases (namespaceId)

### ISSUE THREAD

**Issue by evias** [nemtech/catapult-server#29]
## Expected behaviour

When creating HashLock transactions, it should be possible to specify an **alias** (namespaceId) instead of the *mosaicId* of funds to be locked.

## Actual behaviour

Currently if using aliases' namespaceId when creating a HashLock transaction, broadcast will throw a `TransactionStatusError` with error message: `Failure_Invalid_Mosaic_Id `

**Comment by gimer** [comment]
HashLockObserver seems to correctly resolve mosaic id even in cow (per https://github.com/nemtech/catapult-server/blob/18129e864e2f4e6633aa1db1b8245d6eb6cc6478/plugins/txes/lock_hash/src/observers/HashLockObserver.cpp#L40)

**Comment by evias** [comment]
Thanks for looking into it, the exact error that is thrown seems to be : **Failure_LockHash_Invalid_Mosaic_Id**

This is the error I get with following transaction payload:

```
B00000002EBE3F3546D8077DFDDC046CC6073BF66965410D64A7DCF4FD163B826E85C5A02870E6D782632160ED85606AA7BB6C876E368FD362F002E3E592B509177B86026BACD4C2CE09260554CC027FE073CA5C6B764644660DB9E16AC671E1F700F2B6019048410000000000000000A33B5B251600000044B262C46CEABB858096980000000000E80300000000000053C20029A2F35ADDDC6A5E2B3EDC6F7C7A2DAC67CF119AF283B1E7DB5729751A
```

This payload is a HashLock Transaction with **namespaceId** for `cat.currency` which results in `44B262C46CEABB85` on my private network.

**Now when I send the same transaction but replace the namespaceId by a mosaicId** - retrieving the mosaicId from the network with NamespaceHttp.getLinkedMosaicId() - this transaction gets through.

This is the payload with mosaicId instead of namespaceId:

```
B0000000C00A68609D92D2A038BDB4EA76CBF236A84FC294810EDCDA74225A2E4A262C611DDAE73DFC90F681913F65AA56CD0AA0D046014358EDAC147B47BEF6D32160006BACD4C2CE09260554CC027FE073CA5C6B764644660DB9E16AC671E1F700F2B601904841000000000000000080B55E251600000049E15CEDD2F865778096980000000000E803000000000000B59C0DD7219C91CC6401C4943F993A98E87FA8B395FCE6AFD873CF1AC1D5ADE7
```

In this payload, the mosaicId that the `cat.currency` resolves to is `49E15CEDD2F86577`. 

**Would this problem of validation maybe originate from catapult-rest ?**

The source code that i am trying to run is a nem2-cli-_like_ software package, the exact hashlock command is located at: https://github.com/evias/nem2-sandbox/blob/master/src/commands/transaction/hashlock.ts#L116

The application makes use of private keys located in the `src/base-command.ts` file.

**Comment by Jaguar0625** [comment]
Fixed in dragon (95e0c6a).
